### PR TITLE
2219 fetch zeebe broker version for et

### LINKED ISSUE #2219: The engine version shall be logged via E.T. after each deployment
> **This is an epic issue which may be broken down into multiple tasks, if needed**

__What should we do?__

We would like to better understand to which engine a given diagram is deployed.

In particular, we would like to update the deployment event logging to E.T. as following:
Property Name | Property Definition | Data Type | Sample Values
-- | -- | -- | --
diagramType | type of diagram that was deployed | String | bpmn, dmn
diagramMetrics | *various* | JSON |  *various*
outcome | result of the deployment | String | success, failure
error | type of error (if outcome is failure) | String | DIAGRAM_PARSE_ERROR
context | where was the deployment triggered from | String | deploymentTool
**new:** engine | which engine was deployed to | String | camunda, zeebe
**new:** engineVersion | which engine version was deployed to | String | 7.15.0

In particular we need to:

- [x] Validate and finalize the `Property Name` (`engine` and `engineVersion`)
- [x] Validate with Engine teams which APIs to use to fetch the endpoint version
- [ ] Update the [Telemetry documentation](https://github.com/camunda/camunda-modeler/tree/develop/docs/telemetry)

__Why should we do it?__
This will help to better understand what engine versions is deployed to and hence optimize the Modeler experience to it.

__Context__
[Telemetry documentation](https://github.com/camunda/camunda-modeler/tree/develop/docs/telemetry)

### PR DESCRIPTION
**Pull request by MaxTru**
closes #2219

### CODE REVIEW

**Comment by MaxTru** [comment]
@volkergersabeck with this PR the telemetry data for deplyoments would look like this:

```json
{
  "event": "deployment",
  "diagramType": "[bpmn or dmn]",
  "deployment": {
    "outcome": "[success or failure]",
    "context": "[deploymentTool or startInstanceTool]",
    "targetType": [e.g., "camundaCloud"] // based on bpmn XML
  },
  "deployedTo": { // new
    "executionPlatform": "[camunda or zeebe]", // new
    "executionPlatformVersion": "[version deployed to]" // new
  } // new
}
```

Please confirm this is as desired.

File: client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js [hunk 1]
```diff
@@ -712,6 +774,91 @@ describe('<DeploymentPlugin> (Zeebe)', () => {
...
+      // given
+      const deploymentResult = {
+        success: false,
+        response: {}
+      };
+
+      const getTopologyResult = {
+        success: true,
+        response: {
+          gatewayVersion: '1.33.7'
```
Comment (pinussilvestrus) [inline client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:788]: I like this test data 😄

**Comment by MaxTru** [comment]
Review with Volker:

* `executionPlatform` values should be `Camunda Platform` or `Camunda Cloud`

File: client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js (hunk 1 above)
Comment (barmac) [reply client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:788]: Test'd.

File: app/test/spec/zeebe-api-spec.js [hunk 2]
```diff
@@ -682,6 +682,354 @@ describe('ZeebeAPI', function() {
...
+        });
+
+        const parameters = {
+          endpoint: {
+            type: 'selfHosted'
+          }
+        };
+
+        // when
+        const result = await zeebeAPI.checkConnection(parameters);
```
Comment (barmac) [inline app/test/spec/zeebe-api-spec.js:789]: Why do we call `checkConnection` and not `getTopology` in this describe block?

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js [hunk 3]
```diff
@@ -37,6 +37,8 @@ import DeploymentPluginValidator from './DeploymentPluginValidator';
 
 const DEPLOYMENT_CONFIG_KEY = 'zeebe-deployment-tool';
 
+const ET_EXECUTION_PLATFORM_NAME = 'Camunda Cloud';
```
Comment (barmac) [inline client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:40]: This assumes that all Zeebe deployments go to Camunda Cloud. How about self-hosted Zeebe instances?

File: app/test/spec/zeebe-api-spec.js (hunk 2 above)
Comment (MaxTru) [reply app/test/spec/zeebe-api-spec.js:789]: Thanks! This was by mistake

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 3 above)
Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:40]: For me `Camunda Cloud` and `Camunda Platform` are the new to-be-used names for `Camunda Runtime` and `Zeebe` respectively. @volkergersabeck correct me if wrong.

Note that there is still the [target type field](https://github.com/camunda/camunda-modeler/blob/develop/client/src/plugins/usage-statistics/event-handlers/DeploymentEventHandler.js#L132) which would either be `camundaCloud` or `selfHosted`. So can still differenciate that.

Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:40]: Update: I talked to @volkergersabeck. He will think about this considering also different places that we have, and get back to us.

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js [hunk 4]
```diff
@@ -145,11 +147,18 @@ export default class DeploymentPlugin extends PureComponent {
     // (4) deploy
     const deploymentResult = await this.deployWithConfig(savedTab, config);
 
-    // (5) notify interested parties
+    // (5) fetch version deployed as contextual information
+    const getTopologyResult = await this.getTopology(config);
+
+    const { gatewayVersion } = getTopologyResult.response;
+    options.gatewayVersion = gatewayVersion;
```
Comment (barmac) [inline client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:154]: I had another look at this today and I think that the deployment plugin does not need to know anything about topology. I'd rather expect that we call the backend to `getEngineVersion` and expect to get the engine version as a result. So no topology, no gateway version as these are the implementation details of Zeebe API.

Comment (barmac) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:154]: So if the names change in the future, we touch only `zeebe-api` on the backend and not both the backend and the client.

Comment (barmac) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:154]:
```suggestion
    // (5) fetch version deployed as contextual information
    options.engineVersion = await this.getEngineVersion(config);
```

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js [hunk 5]
```diff
@@ -173,6 +182,16 @@ export default class DeploymentPlugin extends PureComponent {
...
 
+  getTopology(config) {
+    const {
+      endpoint
+    } = config;
+
+    const zeebeAPI = this.props._getGlobal('zeebeAPI');
+
+    return zeebeAPI.getTopology(endpoint);
+  }
```
Comment (barmac) [inline client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:193]: Consequently,
```suggestion
  getEngineVersion(config) {
    const {
      endpoint
    } = config;

    const zeebeAPI = this.props._getGlobal('zeebeAPI');

    return zeebeAPI.getEngineVersion(endpoint);
  }
```

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 4 above)
Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js]: I see the point.

I adjusted it in https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1.

I am not 100% sure here (but it is also not critical):

* Now `zeebe-api.js` is not only a rather stupid wrapper around the `zeebe-node-api`, but it actually extracts information from it. This is what I don't like
* As an alternative 1 we should keep zeebe-api#getTopolgy and wrap around that again. However, that increases the codeBase, while not adding so much value
* (Or we go back to the initial solution without https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1)

At the end of the day: with and without https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1 the solution is good from my perspective. Happy to discuss if you want.

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 5 above)
Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js]: See https://github.com/camunda/camunda-modeler/pull/2237#discussion_r625134679

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 3 above)
Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:40]: @barmac (also see slack #modeling-team). Following the discussion with volker using product names is the desired state. So this should be fine

**Comment by MaxTru** [comment]
Hi @barmac, all comments are addressed, can you please re-review? Thanks

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 4 above)
Comment (barmac) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js]: I like the new solution more. I still don't know what "gateway" is, but I think it's overall much better than having just a stupid wrapper. Note that we already interpret the errors in `zeebe-api.js`.

**Pull request merged**